src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/SliderRows/SliderPanel/SliderPanel.tsx

```diff
@@ -0,0 +1,144 @@
+import { useState, useEffect } from 'react';
+import { useFormContext, FieldValues, Path } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import { Slider } from '@mui/material';
+
+import { UploaderUiType, Uploader, Table, UiType } from 'shared/components';
+import { InputController } from 'shared/components/FormComponents';
+import { theme, StyledFlexTopCenter } from 'shared/styles';
+
+import { Header } from './Header';
+import { SliderPanelProps } from './SliderPanel.types';
+import {
+  StyledSliderPanelContainer,
+  StyledInputContainer,
+  StyledScoresContainer,
+} from './SliderPanel.styles';
+import { getHeadCells, getTableRows } from './SliderPanel.utils';
+import { SLIDER_VALUE_HEAD_ROWS, SLIDER_LABEL_ROWS } from './SliderPanel.const';
+import {
+  DEFAULT_SLIDER_MIN_NUMBER,
+  SLIDER_LABEL_MAX_LENGTH,
+} from '../../../ItemConfiguration.const';
+
+export const SliderPanel = <T extends FieldValues>({
+  name,
+  label,
+  isMultiple,
+  onRemove,
+}: SliderPanelProps<T>) => {
+  const [isExpanded, setIsExpanded] = useState(true);
+
+  const { t } = useTranslation('app');
+
+  const { control, watch, setValue, getValues } = useFormContext();
+
+  const { id, min, max, scores } = watch(name);
+
+  useEffect(() => {
+    const { max, scores } = getValues(name);
+
+    const scoresQuantity = max - min + 1;
+
+    if (scores?.length > scoresQuantity) setValue(`${name}.scores`, scores?.slice(1));
+
+    if (scores?.length < scoresQuantity)
+      setValue(`${name as string}.scores`, [Math.min(...scores) - 1].concat(scores));
+  }, [min, name, getValues, setValue]);
+
+  useEffect(() => {
+    const { min, scores } = getValues(name);
+
+    const scoresQuantity = max - min + 1;
+
+    if (scores?.length < scoresQuantity)
+      setValue(`${name}.scores`, scores.concat(Math.max(...scores) + 1));
+
+    if (scores?.length > scoresQuantity)
+      setValue(`${name}.scores`, scores?.slice(0, scoresQuantity));
+  }, [max, name, getValues, setValue]);
```

Comment:
why do you need two duplicated use effects with the same dependency array?

Reply:
pls minimize dep arrays

Reply:
> why do you need two duplicated use effects with the same dependency array?

2 use effects are for different usage with not the same dependencies. The first one is for changing 'min' variable, the second one - 'max' variable
![image](https://user-images.githubusercontent.com/4319881/224952733-50ea580b-8dc7-4f4f-a3ff-af24b0042d44.png)
The idea is in saving previous user input with scores. If 'Min value' changes, the left border of Scores changes (without losing previously entered scores for other values), and the same for 'Max value'. 
It isn't possible to combine them into one, because in the body of useEffect we can't define which property was changed: min or max


> pls minimize dep arrays

As mentioned in React Docs "Make sure the array includes all values from the component scope (such as props and state) that change over time and that are used by the effect. Otherwise, your code will reference stale values from previous renders". So we do need pass all changeable over time variables which are used in useEffect's body (including functions from custom hooks)

Reply:
Make sure it doesn't cause problems like additional renders. Often adding all the funcs requires additional optimizations